### PR TITLE
DB-9553 Do not purge tombstone if antitombstone has the same timestamp

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

File: hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java [hunk 1]
```diff
@@ -44,7 +61,7 @@ private boolean isSorted(List<Cell> list) {
         Cell previous = iter.next();
         while(iter.hasNext()) {
             Cell next = iter.next();
-            if (KeyValue.COMPARATOR.compare(next, previous) < 0) {
+            if (CellComparator.getInstance().compare(next, previous) < 0) {
                 return false;
             }
             previous = next;
```
Comment (jyuanca) [inline hbase_storage/src/main/java/com/splicemachine/si/impl/server/SICompactionStateMutate.java:67]: Remove isSorted if it's no longer used.

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1016/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1025/

**Pull request merged**